.pre-commit-config.yaml

```diff
@@ -19,6 +19,14 @@ repos:
       types: [python]
       pass_filenames: false
 
+- repo: https://github.com/pre-commit/mirrors-mypy
+  rev: 'v1.16.1'
+  hooks:
+    - id: mypy
+      name: Check static typing
+      entry: mypy src
```

Comment:
I think we should include the tests folder here. I think using mypy we should have caught the incorrectly parsed docstring to type hint for `references_to_dataset`, we have tests for all input cases https://github.com/arviz-devs/arviz-base/blob/main/tests/test_reorg.py#L144